crates/sim2h/src/lib.rs

```diff
@@ -1123,17 +1166,42 @@ impl Sim2h {
                 }
             }
 
-            let payload: Opaque = msg.clone().into();
-
             match self.open_connections.get_mut(&uri) {
                 None => {
                     error!("FAILED TO SEND, NO ROUTE: {}", uri);
                     return;
                 }
-                Some((uuid, _con, outgoing_send)) => {
+                Some(OpenConnectionItem {
+                    version,
+                    uuid,
+                    job: _,
+                    sender: outgoing_send,
+                }) => {
                     open_lifecycle("send", uuid, &uri);
-                    if let Err(_) = outgoing_send.send(payload.as_bytes().into()) {
-                        self.disconnect(&uri);
+
+                    if (version > &mut 1)
```

Comment:
I have pretty strong reservations about forking our version support while we're still in a highly iterative development cycle. The debugging complexity could easily / quickly become untenable if we have to rationalize how different versions are interacting.

My vote would be to:

- capture the Cargo.toml version as a constant within the sim2h lib.
- sim2h_worker and hc_cli can pass that constant in when connecting to sim2h_server
- sim2h_server will reject any connections from a version that doesn't match

Reply:
If we're worried about breaking alpha tester patterns, we could potentially include this version in the url scheme for sim2h servers, so we could keep one or two previous versions running as people upgrade / transition.

Reply:
I agree with this in general, however right now it kills holoscape 0.0.7 without having this check.  Happy to drop the check and code quickly.   Also, the problem with the Cargo.toml version may be moving too slowly?

Reply:
Yeah, if this is an emergency hotfix to break something, all good :+1:

I'm mostly concerned with mitigating bugs reported from the wild via the alpha test group... in which case they should only ever get new code when a Cargo.toml version is bumped, right?

If we also want to make sure devs aren't confused for CI runs / etc... one thing we did at TWC in our publish script was have it commit an additional version bump to the develop branch that had an odd numbered patch version, so we knew those were dev versions, and only released versions with even patch versions.